src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_format.py

```diff
@@ -71,7 +71,8 @@ def transform_file_output(result):
     more clearly distinguishes between files and directories. """
     new_result = []
 
-    for item in result.get('items', [result]):
+    iterable = result if isinstance(result, list) else result.get('items', result)
```

Comment:
Returning value from the custom command could be a list instead of a `ListGenerator`. That is case when directories are filtered.